planner/core/collect_column_stats_usage.go

```diff
@@ -133,119 +152,152 @@ func (p *LogicalUnionAll) updateColMapAndAddPredicateColumns(c *predicateColumnC
 	}
 }
 
-func (c *predicateColumnCollector) collectFromPlan(lp LogicalPlan) {
+func (ds *DataSource) addHistNeededColumns(c *columnStatsUsageCollector) {
```

Comment:
Can we change function signature to `func (c *columnStatsUsageCollector) funcName(ds *DataSource)`?

Reply:
done.